src/zenml/materializers/default_materializer_registry.py

```diff
@@ -71,10 +72,25 @@ def __getitem__(self, key: Type[Any]) -> Type["BaseMaterializer"]:
         if key in self.materializer_types:
             return self.materializer_types[key]
         else:
-            raise KeyError(
-                f"Type {key} does not have a default `Materializer`! Please "
-                f"specify your own `Materializer`."
-            )
+            compatible_subclasses = {
+                self.materializer_types[t]
+                for t in self.materializer_types
+                if issubclass(key, t)
+            }
+            if len(compatible_subclasses) == 1:
+                return compatible_subclasses.pop()
```

Comment:
Any reason to use `.pop()` instead of `[0]` here? it's mutating the list in place, but it's not like that is ever reused again anyway, so it should be garbage collected.

Reply:
`compatible_superclass` here is a set and therefore it is not a subscriptable object. That's why decided to use the `pop` method. Not sure how to improve the garbage collection though.